pkg/core/blockchain.go

```diff
@@ -1460,6 +1441,27 @@ func (bc *Blockchain) verifyHashAgainstScript(hash util.Uint160, witness *transa
 		initMD = cs.Manifest.ABI.GetMethod(manifest.MethodInit)
 	}
 
+	v.LoadScriptWithFlags(verification, smartcontract.NoneFlag)
+	v.Jump(v.Context(), offset)
+	if initMD != nil {
+		v.Call(v.Context(), initMD.Offset)
```

Comment:
We really need to test this sequence, I'm not sure it's correct. We're supposed to run invocation script first and then deal with the contract, but this initialization part makes things more messy.

Reply:
I think it is already tested in `Contract` tests for `verifyHashAgainstScript`.
In C# they also first check then load everything https://github.com/neo-project/neo/blob/master/src/neo/SmartContract/Helper.cs#L132

Reply:
OK, the `Call` really just sets up another context, so it should be fine, we have (invocation-init-verify) on invocation stack and then run it, looks fine.